lib/internal/errors.js

```diff
@@ -999,7 +999,20 @@ E('ERR_OUT_OF_RANGE',
     assert(range, 'Missing "range" argument');
     let msg = replaceDefaultBoolean ? str :
       `The value of "${str}" is out of range.`;
-    msg += ` It must be ${range}. Received ${input}`;
+    let received;
+    if (Number.isInteger(input) && Math.abs(input) > 2 ** 32) {
+      received = String(input).replace(/(\d)(?=(\d\d\d)+(?!\d))/g, '$1_');
```

Comment:
This might be a fairly heavyweight way of inserting underscores. (It's also prone to catastrophic backtracking but that isn't really an issue for small inputs; try `'0'.repeat(2**16)` though.)

What's more, it clobbers `RegExp.$1`, `RegExp.$_`, etc. That's undesirable because it's observable by user code (ref: #20549)

Reply:
That's true but it's very unlikely that such values would end up in here. But I'll write a small helper instead.